dask_kubernetes/core.py

```diff
@@ -14,19 +17,186 @@
     yaml = False
 
 import dask
-from distributed.deploy import LocalCluster, Cluster
+from distributed.deploy import SpecCluster, ProcessInterface
 from distributed.comm.utils import offload
-import kubernetes
+from distributed.utils import Log, Logs
+import kubernetes_asyncio as kubernetes
 from tornado import gen
 
-from .objects import make_pod_from_dict, clean_pod_template
+from .objects import (
+    make_pod_from_dict,
+    make_service_from_dict,
+    clean_pod_template,
+    clean_service_template,
+)
 from .auth import ClusterAuth
-from .logs import Log, Logs
 
 logger = logging.getLogger(__name__)
 
 
-class KubeCluster(Cluster):
+class Pod(ProcessInterface):
+    """ A superclass for Kubernetes Pods
+    See Also
+    --------
+    Worker
+    Scheduler
+    """
+
+    def __init__(self, core_api, pod_template, namespace, loop=None, **kwargs):
+        self.pod = None
+        self.core_api = core_api
+        self.pod_template = copy.deepcopy(pod_template)
+        self.namespace = namespace
+        self.name = None
+        self.loop = loop
+        self.kwargs = kwargs
+        super().__init__()
+
+    @property
+    def cluster_name(self):
+        return self.pod_template.metadata.labels["dask.org/cluster-name"]
+
+    async def start(self, **kwargs):
+        self.pod = await self.core_api.create_namespaced_pod(
+            self.namespace, self.pod_template
+        )
+
+        while self.pod.status.phase == "Pending":
+            self.pod = await self.core_api.read_namespaced_pod(
+                self.pod.metadata.name, self.namespace
+            )
+            await asyncio.sleep(0.1)
```

Comment:
I don't suppose `kubernetes_asyncio` provides any sort of event that triggers when the status changes that we could use to avoid polling?

Reply:
I don't think so.